### DISCUSSION TITLE
Closing tooltip

### DISCUSSION THREAD

**Discussion by sapsari** [microsoft/vscode-discussions#1714] [Extension Development QnA] [answered]
How do I close the tooltip from the code?

My extension has inlay hints, and inlay hints have tooltips. Was able to add commands to the tooltips, so tooltips are clickable. When I click the tooltip and the command gets executed, the tooltip is still visible. I want the tooltip to be closed after clicking, how to achieve that with minimal impact, without effecting other windows etc?

Tried these below, none working so far
```
vscode.commands.executeCommand('workbench.action.closePanel');
vscode.commands.executeCommand('hideHover');
vscode.commands.executeCommand('workbench.action.focusActiveEditorGroup');
```

Also the command replaces some text in the text editor first, then refreshes the inlay hints

**Comment by robconery** [comment]
@isidorn do you have a suggestion here?

**Reply by isidorn** [reply]
Not possible via extensions. The hide/show behavior of the tooltip is handled by vscode. 
Can you share a gif showing the undesired behavior you are seeing.

**Comment by sapsari** [comment]
![2024-10-02_16-14-05-ezgif com-optimize](https://github.com/user-attachments/assets/b99aeb1d-0fbe-4acf-a66b-7f939746c10a)

You see two inlay hints ("Default" and "Chest Front-Back") (also inlay hints are provided by the extension)

When I hover cursor over the inlay hint, it's tooltip is displayed. Then I click on an item of the tooltip, and a command is executed.

The command immediately replaces text ("0" is replaced with "4"), and it refreshes the inlay hints afterwards ("Default" is replaced by "Water")

But as you can see, the tooltip is still open/visible after these operations.

I want the tooltip to be closed/invisible after I click on the tooltip (either before or after the command is executed)

Guess there's no API call for that, but is there a recommenced hack to achieve that, such as shifting focus to text editor window or simulating escape key press?

**Reply by gjsjohnmurray** [reply, marked as answer]
Sounds like https://github.com/microsoft/vscode/issues/229906

**Reply by robconery** [reply]
Hi @sapsari is the linked issue relevant to you?

**Reply by sapsari** [reply]
@robconery They are very similar issues. Tbh, I don't know if tooltip hovers and preview hovers use the same codebase, so a fix for that issue may or may not apply here.

**Reply by burkeholland** [reply]
Let's track in https://github.com/microsoft/vscode/issues/229906. Can you please add your scenario and if resolution doesn't solve it, we can create a separate issue. Thank you @gjsjohnmurray!
